### DISCUSSION TITLE
does it work with ng2-chart 3 ?

### DISCUSSION THREAD

**Discussion by tommy86r** [chartjs/chartjs-plugin-annotation#784] [Q&A] [answered]
I'm trying to use **chartjs-plugin-annotation v2** with **ng-chart v3** but it doesn't seem to work.
I created an example on [stackblitz](https://stackblitz.com/edit/labs-ng2-charts-3-kyuwre)

It's a simple project, I installed chart.js, ng2-chart and chartjs-plugin-annotation.

So, I registered the plugin

```
constructor() {
    Chart.register(annotationPlugin);
}
```
And I used it in a chart:

```
options = {
  scales: {
      y: {
        beginAtZero: true,
      },
    },
    plugins: {
      annotation: {
        annotations: [
          {
            type: 'line',
            borderColor: 'black',
            borderWidth: 3,
            scaleID: 'y',
            value: 50,
          },
        ],
      },
    },
  };
```

I get the following error:

_Type '{ scales: { y: { beginAtZero: boolean; }; }; plugins: { annotation: { annotations: { type: string; borderColor: string; borderWidth: number; scaleID: string; value: number; }[]; }; }; }' is not assignable to type '_DeepPartialObject<CoreChartOptions<keyof ChartTypeRegistry> & ElementChartOptions<keyof ChartTypeRegistry> & PluginChartOptions<...> & DatasetChartOptions<...> & ScaleChartOptions<...>>'._

**Comment by stockiNail** [comment, marked as answer]
@tommy86r I wasn't able to open your link. Anyway I used another sample, adding the annotation:

https://stackblitz.com/edit/labs-ng2-charts-3-eesdlj?file=src/app/time-series/time-series.component.ts

and it seems working

**Reply by tommy86r** [reply]
Thanks mate. I updated the link with the correct one. Anyway, based on your example I found the problem... the Type for options was missing:

_options **: ChartOptions<'line'>** = { ... }_
